### PR TITLE
#10 #11 Add bot log functionality

### PR DESCRIPTION
**Pull request by Kriptonic**
- Add log_channel column to guilds table
 - Add command /set_log_channel
 - Add command /disable_log_channel
 - Add a logging function to src/logger.py
 - Add log message output to various actions
 - Add NONE to DatabaseOperation enum

### CODE REVIEW

**Comment by Kriptonic** [comment]
I was thinking for the future that we could have a method for logging that only takes the `guild_id` and a `LogMessageType`, then the method could translate the `LogMessageType` into the relevant text, this would also pave the way for supporting alternative languages, although that belongs on a wish list somewhere and not in our minimum viable product. The logging file has space to support extra functionality which is why it has been added to a file of its own for now.

File: src/data/table/database.py [hunk 1]
```diff
@@ -9,8 +9,12 @@ def pg_timestamp(timestamp: datetime):
 
 
 class DatabaseOperation(Enum):
-    ADDED = 0
-    EDITED = 1
+    # No operation was performed (the record didn't need updating)
```
Comment (Snowcaloid) [inline src/data/table/database.py:12]: In Python, you can write document strings for any element.
You can do this  by typing """my documentation""" under the element.
This also adds the documentation in the IDE if it's supported

Comment (Snowcaloid) [reply src/data/table/database.py:12]: Example: 
`class DatabaseOperation(Enum):`
`    NONE = 0,  'No operation was performed (the record didn't need updating)'`
or
`class DatabaseOperation(Enum):`
`    NONE = 0`
`    """No operation was performed (the record didn't need updating)"""`

**Review by Snowcaloid — CHANGES_REQUESTED** [review]
Looks great! Just a few changes that I'd like. I left the comments on the relevant topics.
Could you also add logging to the buttons (src\buttons.py)?
It'd be extra useful to know when people assign roles to themself or remove them, and even more so if the PL signups were logged as well.

**Pull request closed without merging**